commons/src/test/java/org/sonarsource/analyzer/commons/collections/AVLTreeTest.java

```diff
@@ -499,4 +500,40 @@ public void test_stack_stream() {
     Stream<Integer> stream = stack.stream();
     assertThat(stream).containsExactly(3, 2, 1);
   }
+

```

Comment:
Nitpick: How about a test for a more complex set that mixes objects with and without the same hash code?

Reply:
Done.